chore: Fix downloads script

marikaner opened the pull request:
Now that the version 1 packages are deprecated, the DOM parsing didn't work anymore. AFAIK you can only get the deprecated versions through clicking on the page, there is no URL for it. When researching, I found that the npm page does not reload, therefore the data had to be available from the beginning. I found it on the window object and therefore switched to puppeteer for this.

marikaner commented on scripts/npm-stats.ts:
```diff
@@ -1,25 +1,15 @@
-  );
+async function getVersionsDownloads(page, packageName) {
+  await page.goto(`https://www.npmjs.com/package/${packageName}`);
+  // @ts-ignore
```
```suggestion
  // window as a global variable is not recognized by TypeScript, therefore ignore checks for this line
  // @ts-ignore
```

FrankEssenberger reviewed (approved):
One question. Does `npm` not have a API to get this data? Fetching it from the UI feels odd, but I guess this has been investigated.

florian-richter commented:
> One question. Does `npm` not have a API to get this data? Fetching it from the UI feels odd, but I guess this has been investigated.

It is currently not available but I requested it a while back: https://github.com/npm/feedback/discussions/647

florian-richter commented:
Consider merging this PR to fix the issue with the v12 tests: https://github.com/SAP/cloud-sdk-js/pull/2547 (and adjusting the required checks due to a slight renaming of the v14 tests)

marikaner commented:
> Consider merging this PR to fix the issue with the v12 tests: #2547 (and adjusting the required checks due to a slight renaming of the v14 tests)

As the tests are still failing here, I just updated the downloads job to v14 and added `--ignore-engines` everywhere
